### PR TITLE
Expand PVC instead of deleting it

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- We need pre-upgrade-job since we are changing the PVC size which is immutable hence helm upgrade would fail. So the job would delete the statefulset before upgrading.
- Also added migration guide.

**Related issue(s)**

#11100

### CODE REVIEW

File: docs/migration-guides/1.22-1.23.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+### Out of box Loki volume size
+
+However if the persistence volume claim is out of box setting use following command to expand the volume:
+
+```
+kubectl delete statefulsets.apps -n kyma-system logging-loki
+
+kubectl patch persistentvolumeclaims -n kyma-system storage-logging-loki-0 --patch '{"spec": {"resources": {"requests": {"storage": "30Gi"}}}}'
+```
+> Volume exapansions are only possible if field `allowVolumeExpansion` is set to true. Please look [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims) for more info.
```
Comment (skhalash) [inline docs/migration-guides/1.22-1.23.md:34]:
```suggestion
Volume expansions are only possible if the field `allowVolumeExpansion` of the corresponding storage class is set to true. Please look [here]
```

File: docs/migration-guides/1.22-1.23.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Migrate from 1.22 to 1.23
+
+## Increase the Loki volume size
+In Kyma 1.23.0, the size of the persistent volume claim has been increased from 10Gi to 30Gi
```
Comment (skhalash) [inline docs/migration-guides/1.22-1.23.md:4]: I'd rephrase it a bit since a PVC doesn't have a size, it requests a volume of a certain size. 

``` suggestion
In Kyma 1.23.0, the Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi
```

Comment (skhalash) [reply docs/migration-guides/1.22-1.23.md:4]: The same wording ( the PVC size) is used below

**Review by skhalash — CHANGES_REQUESTED** [review]

**Review by skhalash — APPROVED** [review]
LGTM!

File: docs/migration-guides/1.22-1.23.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+However if the persistence volume claim is out of box setting use following command to expand the volume:
+
+```
+kubectl delete statefulsets.apps -n kyma-system logging-loki
+
+kubectl patch persistentvolumeclaims -n kyma-system storage-logging-loki-0 --patch '{"spec": {"resources": {"requests": {"storage": "30Gi"}}}}'
+```
+>Volume expansions are only possible if the field `allowVolumeExpansion` of the corresponding storage class is set to true on the storage class. Please look [here]
+
+If the above process fails then the persistent volume claim template needs to be deleted. Use following command:
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md]:
```suggestion
If the process fails, delete the persistent volume claim template with the following command:
```

Which process - the one under "Out-of-box Loki volume size" or the one under "Custom Loki volume size" ?  Additionally, can we make clear how the users see that the operation wasn't successful and that they must use the following fix?

File: docs/migration-guides/1.22-1.23.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+```
+kubectl delete statefulsets.apps -n kyma-system logging-loki
+
+kubectl delete persistentvolumeclaims -n kyma-system storage-logging-loki-0
+```
+> This action will delete the existing data.
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:14]: Can we add the information how the users see that the overall operation was successful? Either here, at the end of the entire document, or after the respective instructions that work if there's no error.

And what is the next step? Line 7 announces that the instructions are a preparation for the upgrade, so can we say that the users can now proceed to do the upgrade to 1.23.0?

**Review by NHingerl — CHANGES_REQUESTED** [review]

**Comment by NHingerl** [comment]
Basically, if I understand it correctly, what we're trying to say is:

> With Kyma 1.23.0, default Loki volume size changes from 10Gi to 30Gi.
> Before you upgrade the Kyma version, you must set the desired persistent volume size to 30Gi:
> 
> ## Instructions if your current configuration uses more than than 30Gi
> 1.
> 2.
> 3.
> 
> ## Instructions if your current configuration uses less than 30Gi
> 1.
> 2.
> 3.
>
> Check whether <desired outcome>. Tada, now you can upgrade to Kyma 1.23.0!
> 
> If you get the error message "blabla" (or <some other indication that the operation failed>), you fix it by deleting the existing data with the following command: ...

File: docs/migration-guides/1.22-1.23.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Migrate from 1.22 to 1.23
+
+## Increase the Loki volume size
+In Kyma 1.23.0, the Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi
+
+## Upgrade process
+During the upgrade process, the pre-upgrade job is patching the volume size so as to increase the size from 10Gi to 30Gi as the retention time for logs has been increased from 1 day to 5 days. However, the upgrade process would succeed if the field `allowVolumeExpansion` is set to `true` for the storage class. Please look [here](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims) for more info.
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:7]:
```suggestion
During the upgrade process, the pre-upgrade job increases the volume size from 10Gi to 30Gi. 
For the upgrade process to be successful, make sure that the field `allowVolumeExpansion` is set to `true` for the storage class. For more information, read the [Kubernetes documentation about Expanding Persistent Volumes Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims).
```

File: docs/migration-guides/1.22-1.23.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+If the flag is not set on the storage class then the size of the volume would not change. Use following command to check the size of the volume
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:9]:
```suggestion
If the flag is not set on the storage class, the size of the volume would not change. Use following command to check the size of the volume:
```

File: docs/migration-guides/1.22-1.23.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+```
+kubectl get persistentvolumeclaims -n kyma-system storage-logging-loki-0
+```
+if the size is still the old size follow the steps below and then trigger update
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:14]:
```suggestion
If the size is still the old size, delete the old data with the following command:
```

File: docs/migration-guides/1.22-1.23.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+```
+kubectl delete statefulsets.apps -n kyma-system logging-loki
+
+kubectl delete persistentvolumeclaims -n kyma-system storage-logging-loki-0
+```
+> This action will delete the existing data.
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:21]:
```suggestion
After the old data is deleted, trigger the Kyma update.
```

File: docs/migration-guides/1.22-1.23.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+If there is custom volume size is set for the loki persistent volume claim, then evaluate if the current size would suffice for the logs for retention time of 5 days.
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:23]:
```suggestion
Make sure that your custom volume size for the Loki persistent volume claim is sufficient for the new retention time of 5 days.
```

I put a similar hint right at the top, so I'm not sure if this is needed here again. Not sure if we want users to evaluate that right away, or at the end.

**Review by NHingerl — CHANGES_REQUESTED** [review]

File: docs/migration-guides/1.22-1.23.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+## Increase the Loki volume size
+In Kyma 1.23.0, the default Loki volume size requested by the persistent volume claim has been increased from 10Gi to 30Gi because the retention time for logs has been increased from 1 day to 5 days. 
+
+> If you are using a custom volume size for the Loki persistent volume claim, we recommend that you evaluate whether that volume size is sufficient to retain the logs for 5 days.
+
+## Upgrade process
+During the upgrade process, the pre-upgrade job increases the volume size from 10Gi to 30Gi. 
+For the upgrade process to be successful, make sure that the field `allowVolumeExpansion` is set to `true` for the storage class. For more information, read the [Kubernetes documentation about Expanding Persistent Volumes Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims).
+
+If the flag is not set on the storage class, the size of the volume would not change. Use following command to check the size of the volume:
```
Comment (NHingerl) [inline docs/migration-guides/1.22-1.23.md:12]:
```suggestion
If the flag is not set on the storage class, the size of the volume would not change. Use the following command to check the size of the volume:
```

**Review by NHingerl — CHANGES_REQUESTED** [review]
Missed this one earlier

**Review by NHingerl — APPROVED** [review]

**Review by skhalash — APPROVED** [review]
